src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettingsTemplatePackageProvider.cs

```diff
@@ -104,13 +104,27 @@ public async Task<IReadOnlyList<CheckUpdateResult>> GetLatestVersionsAsync(IEnum
             return result;
         }
 
+        /// <inheritdoc/>
+        /// <exception cref="ArgumentException">when <paramref name="installRequests"/> has duplicate identifiers for given installer.</exception>
         public async Task<IReadOnlyList<InstallResult>> InstallAsync(IEnumerable<InstallRequest> installRequests, CancellationToken cancellationToken)
         {
             _ = installRequests ?? throw new ArgumentNullException(nameof(installRequests));
             if (!installRequests.Any())
             {
                 return new List<InstallResult>();
             }
+
```

Comment:
I'm not sure about this validation as Provider cannot be sure if Installer supports the duplicates are not. I'm more keen to keep validation only in host.